game/state/shared/aequipment.cpp

```diff
@@ -702,12 +702,19 @@ void AEquipment::fire(GameState &state, Vec3<float> targetPosition, StateRef<Bat
 	auto unit = ownerAgent->unit;
 	auto payload = getPayloadType();
 	Vec3<float> originalTarget = targetPosition;
+	int number_of_shots;
 
 	if (payload->fire_sfx)
 	{
 		fw().soundBackend->playSample(payload->fire_sfx, unit->position);
 	}
 
+	number_of_shots = this->type->burst;
```

Comment:
Should this be clamped against the current ammo?

e.g. `std::min(this->type->burst, this->ammo)` would make sure it only fires at most as many shots are loaded

Reply:
Also, this currently only takes a single ammo out of the clip no matter the size of the burst - is this intended? IE in burst weapons should the "ammo" count actually be the "number of bursts?" Or the number of individual shots?

Reply:
It's by design, but thinking about what you've said, that needs to change

My initial thinking was "It's a burst weapon, just specify the number of bursts as ammo"

However, and the more i think about it, I probably want to treat it as "amount of rounds". This way if i ever allow for multiple fire modes with modded weapons, for example, Single Shot, Burst and Full Automatic, it counts by round and not just how many times you can pull the trigger to whatever result

Thanks for prompting me to think about this :) It makes sense to change it now to save issue in the future

I'll change line 712 to what you suggested (I suspect also i will have to look at vequipment.cpp also)

As to how to get it to deduct only the number of rounds used by the fire mode, how would i do this?